tests/Silverback.Integration.Tests.E2E/Kafka/ProducerTests.cs

```diff
@@ -1594,5 +1594,159 @@ public async Task RawProduce_WithKafkaKeyHeader_KafkaKeySet()
                     .ContainEquivalentOf(new MessageHeader(KafkaMessageHeaders.KafkaMessageKey, "42-42-42"));
             }
         }
+
+        [Fact]
+        public async Task RawProduce_WithMessageIdHeader_TimestampKeySet()
```

Comment:
Why 2 tests? Is the timestamp somehow influenced by the message id or kafka key header?

Also, why only testing `RawProduce`? (Genuine question, there might be a reason.)